crates/sui-core/src/consensus_adapter.rs

```diff
@@ -212,17 +256,37 @@ impl ConsensusAdapter {
         // refreshed frequently enough to make sure this is Byzantine-resistant
     }
 
+    /// This method is called externally to begin reconfiguration
+    /// It transition reconfig state to reject new certificates from user
+    /// ConsensusAdapter will send EndOfPublish message once pending certificate queue is drained
+    pub fn begin_reconfiguration(self: &Arc<Self>) -> SuiResult {
+        let send_end_of_publish = {
+            let reconfig_guard = self.authority.get_reconfig_state_write_lock_guard();
+            self.authority.close_user_certs(reconfig_guard);
```

Comment:
This call would release reconfig_guard. Is that intended?

Reply:
I can't immediately think if this can be bad that we drop guard here, but just to make things uniformly with other places I changed is so that we inspect `pending_certificates` while holding the lock